File: packages/typia/src/transformers/features/http/CreateHttpQueryTransformer.ts
import { HttpQueryProgrammer } from "../../../programmers/http/HttpQueryProgrammer";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace CreateHttpQueryTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "http.createQuery",
      write: HttpQueryProgrammer.write,
    });
}

File: packages/transform/src/features/http/CreateHttpQueryTransformer.go
package http

import (
	corehttp "github.com/samchon/typia/packages/core/src/programmers/http"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var CreateHttpQueryTransformer = createHttpQueryTransformerNamespace{}

type createHttpQueryTransformerNamespace struct{}

func (createHttpQueryTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return corehttp.HttpQueryProgrammer.Write(call.TypeText)
}
